packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FWFScrollViewDelegateHostApi.m

```diff
@@ -0,0 +1,90 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "FWFScrollViewDelegateHostApi.h"
+#import "FWFWebViewHostApi.h"
+
+@interface FWFScrollViewDelegateFlutterApiImpl ()
+// BinaryMessenger must be weak to prevent a circular reference with the host API it
+// references.
+@property(nonatomic, weak) id<FlutterBinaryMessenger> binaryMessenger;
+// InstanceManager must be weak to prevent a circular reference with the object it stores.
+@property(nonatomic, weak) FWFInstanceManager *instanceManager;
+@end
+
+@implementation FWFScrollViewDelegateFlutterApiImpl
+
+- (instancetype)initWithBinaryMessenger:(id<FlutterBinaryMessenger>)binaryMessenger
+                        instanceManager:(FWFInstanceManager *)instanceManager {
+  self = [self initWithBinaryMessenger:binaryMessenger];
+  if (self) {
+    _binaryMessenger = binaryMessenger;
+    _instanceManager = instanceManager;
+  }
+  return self;
+}
+- (long)identifierForDelegate:(FWFScrollViewDelegate *)instance {
+  return [self.instanceManager identifierWithStrongReferenceForInstance:instance];
+}
+
+- (void)scrollViewDidScrollForDelegate:(FWFScrollViewDelegate *)instance
+                          uiScrollView:(UIScrollView *)scrollView
+                            completion:(void (^)(FlutterError *_Nullable))completion {
+  [self scrollViewDidScrollWithIdentifier:[self identifierForDelegate:instance]
+                   uiScrollViewIdentifier:[self.instanceManager
```

Comment:
if instanceManager is weak, do you want to deal with nil case?

Reply:
I don't think `instanceManager` can be nil here. Again, I just followed the pattern of other host API was implemented before me.

The reason why I think `instanceManager` couldn't be nil is because it is strongly referenced by the plugin [here](https://github.com/flutter/packages/commit/313ef61bc1b3b6b41e38573641d46ab49afe40ea#diff-97d58ad121730325019c0434ec1c12c2a47c62e07c6bb09a54e48a619d695601R107)

Reply:
`instanceManager` can be null in an add-to-app scenario. However, Objective-c will just return `nil` and not cause a crash. This is the desired behavior since the plugin holds the only strong reference and we want this to be null when that changes. I will make this clearer when https://github.com/flutter/flutter/issues/134777 is implemented in this language.